src/auro-radio-group.js

```diff
@@ -145,65 +168,27 @@ export class AuroRadioGroup extends LitElement {
         });
       }
     }
+
     if (changedProperties.has('required')) {
       this.items.forEach((el) => {
         el.required = this.required;
       });
     }
-    if (changedProperties.has('error')) {
-      this.validate();
-    }
-  }
 
-  /**
-   * Determines the validity state of the element.
-   * @private
-   * @returns {void}
-   */
-  /* eslint-disable max-statements */
-  validate() {
-    // Validate only if noValidate is not true and the input does not have focus
-    if (this.hasAttribute('error')) {
-      this.validity = 'customError';
-      this.setCustomValidity = this.error;
-    } else if (this.value !== undefined && !this.noValidate) {
-      this.validity = 'valid';
-      this.setCustomValidity = '';
-
-      /**
-       * Only validate once we interact with the datepicker
-       * this.value === undefined is the initial state pre-interaction.
-       *
-       * The validityState definitions are located at https://developer.mozilla.org/en-US/docs/Web/API/ValidityState.
-       */
-      if ((!this.value || this.value.length === 0) && this.required) { // eslint-disable-line no-magic-numbers
-        this.validity = 'valueMissing';
-        this.setCustomValidity = this.setCustomValidityValueMissing;
-      }
-    } else {
-      this.validity = undefined;
-      this.removeAttribute('validity');
+    if (changedProperties.has('error')) {
+      this.validation.validate(this, true);
     }
 
-    if (this.validity && this.validity !== 'valid') {
-      this.isValid = false;
-
-      // Use the validity message override if it is declared
-      if (this.ValidityMessageOverride) {
-        this.setCustomValidity = this.ValidityMessageOverride;
+    if (changedProperties.has('validity')) {
+      if (this.validity && this.validity !== 'valid') {
+        this.items.forEach((el) => {
+          el.setAttribute('error', true);
+        });
+      } else {
+        this.items.forEach((el) => {
+          el.removeAttribute('error');
+        });
```

Comment:
Explain code in comments, it's not obvious why this needs to happen

```suggestion
    if (changedProperties.has('validity')) {
      /** 
       * Since the individual radio buttons render the invalid styles they must each
       * have the `error` attribute applied or removed based on the validity of the
       * auro-radio-group.
       */
      if (this.validity && this.validity !== 'valid') {
        this.items.forEach((el) => {
          el.setAttribute('error', true);
        });
      } else {
        this.items.forEach((el) => {
          el.removeAttribute('error');
        });
```